packages/makeup-hoist-element/src/index.js

```diff
@@ -0,0 +1,79 @@
+let modalEl;
+let modalPlaceholder;
+let containerDiv;
+let prevAriaHiddenSetting;
+let bodyIndexes = [];
+
+function unhoist() {
+    if (modalEl) {
+        if (containerDiv) {
+            const childList = Array.from(containerDiv.childNodes);
+            childList.forEach((child) => {
+                if (child.src === undefined) {
+                    const index = bodyIndexes.shift();
+                    if (index > document.body.childNodes.length) {
+                        document.body.appendChild(child);
+                    } else {
+                        document.body.insertBefore(child, document.body.childNodes[index + 1]);
+                    }
+                }
+            });
+            containerDiv.remove();
+            containerDiv = null;
+            bodyIndexes = [];
+        }
+
+        if (modalPlaceholder) {
+            modalEl.removeAttribute('aria-hidden');
+            // eslint-disable-next-line eqeqeq
+            if (prevAriaHiddenSetting != null) {
+                modalEl.setAttribute('aria-hidden', prevAriaHiddenSetting);
+            }
+            prevAriaHiddenSetting = null;
+            modalPlaceholder.replaceWith(modalEl);
+            // let observers know the element is unhoisted
+            const event = document.createEvent('Event');
+            event.initEvent('unhoist', true, true);
+            modalEl.dispatchEvent(event);
+            modalPlaceholder = null;
+        }
+        modalEl = null;
+    }
+    return modalEl;
+}
+
+function hoist(el) {
+    unhoist();
+    modalEl = el;
+
+    modalPlaceholder = document.createElement('div');
+    modalEl.insertAdjacentElement('beforebegin', modalPlaceholder);
+
+    containerDiv = document.createElement('div');
+    const childList = Array.from(document.body.childNodes);
+    childList.forEach((child, index) => {
```

Comment:
Since childList is not used anywhere else, you can do this as a 1 line
This also works:
`[...document.body.childNodes].forEach((child, index)`, but I'm fine with the `Array.from` syntax. No need to have it in two lines I think 

Also can you explain what is the point of this method? Why are you checking for `child.src`?

Reply:
Well I was separating out the rest of the document.body child nodes from the <script/> tags. I was worried about unintended side effects of moving the location of the <script/> tags. That's also why I keep track of the body indexes - so that I can put everything back in its original location.